src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/InProcess/RazorProjectSystemInProcess.cs

```diff
@@ -19,6 +19,13 @@ namespace Microsoft.VisualStudio.Extensibility.Testing;
 [TestService]
 internal partial class RazorProjectSystemInProcess
 {
+    public async Task<bool> IsCohostingActiveAsync(CancellationToken cancellationToken)
+    {
+        var options = await TestServices.Shell.GetComponentModelServiceAsync<LanguageServerFeatureOptions>(cancellationToken);
```

Comment:
>var options = await TestServices.Shell.GetComponentModelServiceAsync<LanguageServerFeatureOptions>(cancellationToken); [](http://example.com/codeflow?start=8&length=118)

Looks good. Curious why you didn't use an AsyncLazy though, can UseRazorCohostServer change?